OpenRA.Game/Graphics/Minimap.cs

```diff
@@ -70,23 +99,30 @@ static Bitmap AddStaticResources(TileSet tileset, Map map, Ruleset resourceRules
 			{
 				var colors = (int*)bitmapData.Scan0;
 				var stride = bitmapData.Stride / 4;
-				for (var y = 0; y < b.Height; y++)
+				for (var y = 0; y < height; y++)
 				{
-					for (var x = 0; x < b.Width; x++)
+					for (var x = 0; x < width; x++)
 					{
-						var mapX = x + b.Left;
-						var mapY = y + b.Top;
-						if (map.MapResources.Value[new MPos(mapX, mapY)].Type == 0)
+						var uv = new MPos(x + b.Left, y + b.Top);
+						if (map.MapResources.Value[uv].Type == 0)
 							continue;
 
-						var res = resourceRules.Actors["world"].Traits.WithInterface<ResourceTypeInfo>()
```

Comment:
Ugh, this trait lookup was done for every pixel in the minimap. Nice cleanup. :+1: